sound/soc/sof/intel/hda-stream.c

```diff
@@ -229,8 +241,12 @@ int hda_dsp_stream_trigger(struct snd_sof_dev *sdev,
 		hstream->running = true;
 		break;
 	case SNDRV_PCM_TRIGGER_SUSPEND:
-	case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
 	case SNDRV_PCM_TRIGGER_STOP:
+		/* unlock link DMA */
+		stream->link_locked = 0;
```

Comment:
same here, you don't actually unlock with this assignment, you just prepare for it to be unlocked, but when that happens is unclear.

Reply:
@plbossart basically my assumption is this:

1. For a HDA stream, hda_stream_hw_params() would allocate the host DMA channel and hda_link_hw_params() would allocate the link DMA channel. So there will be no issues with coupling/decoupling

2. For an I2S stream for example, hda_stream_hw_params() would allocate the host DMA but there would be no hda_link_hw_params(). So when the START trigger follows, I check if the corresponding link DMA channel is locked or not? If it isnt (which is because this is a I2S playback/capture stream, I set link_locked so that the link DMA channel will not be allocated to the next stream that is started. 

3. Similarly, in the case of hostless streams, there will not be a hda_stream_hw_params() but there will be hda_link_hw_params(). So we would set the stream->opened to true to prevent the host DMA from being assigned to the next stream that is started.